src/RA_StringUtils.cpp

```diff
@@ -83,18 +83,24 @@ std::string Narrow(const std::string& str)
 _Use_decl_annotations_
 std::string& TrimLineEnding(std::string& str) noexcept
 {
-    auto nIndex = str.length();
-    if (nIndex > 0 && str[nIndex - 1] == '\n')
-        nIndex--;
-    if (nIndex > 0 && str[nIndex - 1] == '\r')
-        nIndex--;
-
-    if (nIndex != str.length())
-        str.resize(nIndex);
-
+    auto sCpy{ str };
+    str = TrimLineEnding(std::move_if_noexcept(sCpy));
     return str;
 }
 
+std::string TrimLineEnding(std::string&& str) noexcept
+{
+    auto sRet{ std::move_if_noexcept(str) };
+    
+    // using the .back() and .pop_back() strategy doesn't work
```

Comment:
Firstly, why not?

Secondly, this sort of comment is irrelevant once the code is committed. Someone coming across it in six months will have no idea what the "back/pop_back" strategy was. This is one of the things that should be saved for the review comments.